src/server.cpp

```diff
@@ -93,11 +93,26 @@ void *ServerThread::run()
 {
 	BEGIN_DEBUG_EXCEPTION_HANDLER
 
-	m_server->AsyncRunStep(true);
+	u64 step_ms = m_server->getStepSize() * 1000.0f;
+	u64 last_step = porting::getTimeMs();
+	u64 new_step = old_step;
 
 	while (!stopRequested()) {
+		new_step = porting::getTimeMs();
+		u64 dtime_ms = porting::getDeltaMs(last_step, new_step);
+
+		if (dtime_ms < step_ms) {
+			std::this_thread::sleep_for(std::chrono::milliseconds(
```

Comment:
I don't think this thread should (explicitly) sleep since it also handles packet receiving, which is then going to be delayed.

Reply:
Either way packets are already delayed in the current code, by multiples of the server step.
Packet processing could be moved to a new thread if you'd prefer that.

Reply:
Are they? `AsyncRunStep` gets called every 30ms (timeout of `Receive`) and immediately returns unless a "full" server step can be made. So between the server steps happening, it will check for new packets every 30ms.

If you change it to sleep, `Receive` will only be called between fixed server steps of 100ms(?).

> Packet processing could be moved to a new thread if you'd prefer that.

The server code is single-threaded, synchronization would make a separate thread useless.